src/Microsoft.Health.Dicom.Functions/DurableFunctionsTimedCleanup.cs

```diff
@@ -0,0 +1,81 @@
+{
+    public class DurableFunctionsTimedCleanup
+    {
+        private readonly ReindexErrorConfiguration _cleanupConfig;
+        private readonly Func<DateTime> _getUtcNow;
+
+        public DurableFunctionsTimedCleanup(IOptions<ReindexErrorConfiguration> cleanupOptions)
+            : this(cleanupOptions, () => DateTime.UtcNow)
+        { }
+
+        internal DurableFunctionsTimedCleanup(IOptions<ReindexErrorConfiguration> cleanupOptions, Func<DateTime> getDateTimeUtcNow)
+        {
+            EnsureArg.IsNotNull(cleanupOptions, nameof(cleanupOptions));
+            EnsureArg.IsNotNull(getDateTimeUtcNow, nameof(getDateTimeUtcNow));
+            _cleanupConfig = cleanupOptions.Value;
+            _getUtcNow = getDateTimeUtcNow;
+        }
+
+        [FunctionName("DurableFunctionsTimedCleanup")]
+        public async Task Run([TimerTrigger("*/15 * * * * *")] TimerInfo myTimer,
+            [DurableClient] IDurableOrchestrationClient client,
+            ILogger log,
+            CancellationToken hostCancellationToken)
+        {
+            List<string> orchestrationInstanceIdList = new List<string>();
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            EnsureArg.IsNotNull(client, nameof(client));
+            EnsureArg.IsNotNull(myTimer, nameof(myTimer));
+
+            log.LogInformation("Durable Functions cleanup started at: " + myTimer.ToString());
+            string isPastDue = myTimer.IsPastDue ? "Current function invocation is later than scheduled." : "Function invocation is on schedule.";
+            log.LogInformation(isPastDue);
+
+            EnsureArg.IsNotNull(client, nameof(client));
+
+            // Specify conditions for Function Queries.
+            OrchestrationStatusQueryCondition condition = new OrchestrationStatusQueryCondition
+            {
+                RuntimeStatus = _cleanupConfig.RuntimeStatuses,
+                CreatedTimeFrom = DateTime.MinValue,
+                CreatedTimeTo = _getUtcNow().AddDays(-_cleanupConfig.MinimumAgeDays),
+                ContinuationToken = null
+            };
+
+            do
+            {
+                OrchestrationStatusQueryResult listOfOrchestrators =
+                   await client.ListInstancesAsync(condition, hostCancellationToken);
+                condition.ContinuationToken = listOfOrchestrators.ContinuationToken;
+
+                // Loop through the orchestration instances and purge them.
+                foreach (var orchestration in listOfOrchestrators.DurableOrchestrationState)
+                {
+                    orchestrationInstanceIdList.Add(orchestration.InstanceId);
+                    await client.PurgeInstanceHistoryAsync(orchestration.InstanceId);
+                }
+
+            } while (condition.ContinuationToken != null);
+
+            log.LogInformation($"Durable Functions cleaned up successfully:\n{string.Join("\n", orchestrationInstanceIdList)}");
```

Comment:
Update this to leverage structured logging. And perhaps we can have two elements:
1. The number of instances purged
2. The instance IDs